cx/twitter/server.js

```diff
@@ -0,0 +1,279 @@
+          }
+        });
+  });
+}
+
+function registerSubscription() {
+  request.post(
+      'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+          '/subscriptions.json',
+      {
+        oauth: twitterOAuth,
+      },
+      function(err, resp, body) {
+        if (err) {
+          console.error('Failed to register subscription: ' + err);
+        }
+      });
+}
+
+function listWebhooks(url) {
+  return new Promise((resolve, reject) => {
+    request.get(
+        'https://api.twitter.com/1.1/account_activity/all/webhooks.json',
+        {oauth: twitterOAuth}, function(err, resp, body) {
+          if (err) {
+            console.error('Failed to check webhooks: ' + err);
+            reject(err);
+          }
+          const environments = JSON.parse(body).environments;
+          if (Array.isArray(environments)) {
+            const environment = environments.find((element) => {
+              return element.environment_name === environmentName;
+            });
+            if (environment) {
+              const webhooks =
+                  environment.webhooks.filter((value, index, arr) => {
+                    return value.url === url;
+                  });
+              resolve(webhooks);
+            }
+            resolve([]);
+          }
+        });
+  });
+}
+
+function deleteWebhookById(webhookId) {
+  return new Promise((resolve, reject) =>{
+    request.delete(
+        'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+            '/webhooks/' + webhookId + '.json',
+        {oauth: twitterOAuth}, function(err, resp, body) {
+          if (err) {
+            console.error('Failed to delete webhook: ' + err);
+            reject(err);
+          }
+          resolve();
+        });
+  });
+}
+
+async function deleteWebhooksByUrl(targetUrl) {
+  const webhooks = await listWebhooks(targetUrl);
+  for (webhook of webhooks) {
+    if (webhook.id) {
+      await deleteWebhookById(webhook.id);
+    }
+  }
+}
+
+function deleteSubscription() {
+  request.delete(
+      'https://api.twitter.com/1.1/account_activity/all/' + environmentName +
+          '/subscriptions.json',
+      {oauth: twitterOAuth}, function(err, resp, body) {
+        if (err) {
+          console.error('Failed to delete subscriptions: ' + err);
+        }
+      });
+}
+
+app.post('/', async function(req, res) {
+  if (Array.isArray(req.body.direct_message_events) &&
+      req.body.direct_message_events.length > 0) {
+    const message = req.body.direct_message_events[0];
+    const text = message.message_create.message_data.text;
+    const senderId = message.message_create.sender_id;
+    if (senderId !== twitterId) {
+      var responses = await detectIntentResponse(text, senderId);
+
+      for (let response of responses.queryResult.responseMessages) {
+        if (response.hasOwnProperty('text')) {
+          sendMessage(response.text.text.join(), senderId);
+          res.sendStatus(200);
```

Comment:
It might not make sense to run `res.sendStatus()` for each messages in `detectIntentResponse`. Probably only need to run it once at the end of the handler.

Reply:
Restructured the code so that it runs res.sendStatus() after send all messages.